src/platforms/javascript/common/configuration/tree-shaking/index.mdx

```diff
@@ -36,7 +36,49 @@ To make optional code eligible for tree shaking, you can replace various flags i
 : Replacing this flag with `false` will tree shake all code in the SDK that is related to performance monitoring.
 **Attention:** `__SENTRY_TRACING__` must not be replaced with `false` when you're using any performance monitoring-related SDK features (e.g. `Sentry.startTransaction()`). This flag is intended to be used in combination with packages like `@sentry/next` or `@sentry/sveltekit`, which automatically include performance monitoring functionality.
 
-<PlatformSection notSupported={["javascript.nextjs"]}>
+`__RRWEB_EXCLUDE_CANVAS__`
+
+: Replacing this flag with `true` will tree shake all code in the SDK that is related capturing canvas elements with Session Replay. This is only relevant when using [Session Replay](../../session-replay/). You can enable this flag if you don't have any canvas elements on your page you care to record.
+
+`__RRWEB_EXCLUDE_IFRAME__`
+
+: Replacing this flag with `true` will tree shake all code in the SDK that is related capturing iframe content with Session Replay. This is only relevant when using [Session Replay](../../session-replay/). You can enable this flag if you don't have any iframes on your page you care to record.
+
+`__RRWEB_EXCLUDE_SHADOW_DOM__`
+
+: Replacing this flag with `true` will tree shake all code in the SDK that is related capturing shadow dom elements with Session Replay. This is only relevant when using [Session Replay](../../session-replay/). You can enable this flag if you don't have any shadow dom elements on your page you care to record.
+
+<PlatformSection notSupported={["javascript.nextjs", 'javascript.sveltekit', 'javascript.remix']}>
```

Comment:
Astro too I believe. @Lms24 ?

Reply:
The Astro SDK currently doesn't expose Vite plugin options except for a few hand-picked ones to avoid depending on the plugin's API. I can update the SDK though to include this new API.

Reply:
> (I guess that's the downside of not just passing through plugin options)

Yup that's a bit annoying, but I would say it is worth not having to break.

Stupid thought: We could have like one central layer (in utils or smth) for all the higher level JS SDKs that reexposes certain options. That way we wouldn't have to do this for all SDKs separately. (When we have Astro, Next.js, Sveltekit, and probably a million others soon (luc.js :P))

Reply:
> We could have like one central layer (in utils or smth) for all the higher level JS SDKs that reexposes certain options

Hmm good idea! Let's see how things develop in Astro. Maybe we can extract this layer/API out for the next (!== NextJs) SDK or use it in existing SDKs when we break in v8.